src/lib/classes/access-control-list.js

```diff
@@ -0,0 +1,333 @@
+  }
+
+  /**
+   * @function createsQuad Creates a simple quad object
+   * @param {String} subject Subject of the quad
+   * @param {String} predicate Predicate of the quad
+   * @param {String} object Object of the quad
+   */
+  createQuad = (subject, predicate, object) => ({
+    subject,
+    predicate,
+    object
+  });
+
+  /**
+   * @function createQuadList Creates a N3 quad list to later be parsed as Turtle
+   * @param {Array<String>} modes Array of modes
+   * @param {Array<String> | null} agents Array of webId or null if for everyone
+   */
+  createQuadList = (modes: Array<String>, agents: Array<String> | null) => {
+    const { acl, foaf, a } = ACL_PREFIXES;
+    const subject = `${this.aclUri}#${modes.join('')}`;
+    const originalPredicates = [
+      this.createQuad(subject, `${a}`, namedNode(`${acl}Authorization`)),
+      this.createQuad(subject, `${acl}accessTo`, namedNode(this.documentUri))
+    ];
+    let predicates = [];
+    if (agents) {
+      const agentsArray = Array.isArray(agents) ? agents : [agents];
+      const agentsQuads = agentsArray.map(agent =>
+        this.createQuad(subject, `${acl}agent`, namedNode(agent))
+      );
+      predicates = [...originalPredicates, ...agentsQuads];
+    } else {
+      const publicQuad = this.createQuad(subject, `${acl}agentClass`, namedNode(`${foaf}Agent`));
+      predicates = [...originalPredicates, publicQuad];
+    }
+
+    const quadList = modes.reduce(
+      (array, mode) => [
+        ...array,
+        this.createQuad(subject, `${acl}mode`, namedNode(`${acl}${mode}`))
+      ],
+      predicates
+    );
+
+    return quadList;
+  };
+
+  /**
+   * @function createPermissionsTurtle Creates a turtle looking string with specific permissions
+   * @param {Array<Permissions> | null} permissions Array of permissions to be added to the turtle string
+   * @return {String } A Turtle looking string with all of the necessary permissions
+   */
+  createPermissionsTurtle = (permissions: Array<Permissions>) => {
+    const { DataFactory } = N3;
+    const prefixes = { ...ACL_PREFIXES, '': `${this.aclUri}#`, me: this.owner };
+    const { namedNode, quad } = DataFactory;
+    const writer = new N3.Writer({ prefixes });
+    const quadPermissions = permissions.map(({ modes, agents }) =>
+      this.createQuadList(modes, agents)
+    );
+    const quads = quadPermissions
+      .map(quadItem => {
+        const itemQuads = quadItem.map(({ subject, predicate, object }) =>
+          quad(namedNode(subject), namedNode(predicate), object)
+        );
+        return itemQuads;
+      })
+      .reduce((array, item) => [...array, ...item], []);
+
+    quads.forEach(quad => writer.addQuad(quad));
+    let turtle;
+    writer.end((error, result) => {
+      if (!error) turtle = result;
+    });
+    return turtle;
+  };
+
+  /**
+   * @function createACLFile Creates a file or a container with it specfic set of acls. Assigns READ, WRITE and CONTROL permissions to the owner by default
```

Comment:
Typo, should read "specific"

Reply:
Done

Reply:
Hm, I think looking at it now, the first part should read "Creates a file or container with a specific set of acls"

Reply:
done